File: src/routes/banks.ts


import { Router, Request, Response } from 'express';
import { config } from '../config';
import { getAllBanks, getCacheInfo } from '../services/bankRegistry';

const router = Router();

router.get('/banks', (_req: Request, res: Response) => {
  const banks = getAllBanks();
  const cacheInfo = getCacheInfo();

  res.json({
    status: 'success',
    data: {
      banks: banks.map((b) => ({
        code: b.code,
        name: b.name,
        slug: b.slug,
        active: b.active,
      })),
      last_updated: cacheInfo.lastUpdated,
      total_count: cacheInfo.totalCount,
    },
    metadata: {
      api_version: config.apiVersion,
      timestamp: new Date().toISOString(),
    },
  });
});

export default router;

File: snapaccount-py/routes/banks.py


from fastapi import APIRouter
from config import API_VERSION
from services.bank_registry import get_all_banks, get_cache_info

router = APIRouter()

@router.get("/banks")
async def list_banks():
    """Return the cached bank list."""
    banks = get_all_banks()
    cache_info = get_cache_info()

    return {
        "status": "success",
        "data": {
            "banks": [
                {
                    "code": b["code"],
                    "name": b["name"],
                    "slug": b.get("slug", ""),
                    "active": b.get("active", True),
                }
                for b in banks
            ],
            "last_updated": cache_info["last_updated"],
            "total_count": cache_info["total_count"],
        },
        "metadata": {
            "api_version": API_VERSION,
        },
    }
